core/src/main/java/org/opensearch/sql/expression/function/PrometheusFunctions.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.function;
+
+import static org.opensearch.sql.data.type.ExprCoreType.LONG;
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.FunctionExpression;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+import org.opensearch.sql.expression.env.Environment;
+
+
+/**
+ * Utility Class for prometheus connector specific functions.
+ */
+@UtilityClass
+public class PrometheusFunctions {
+
+  public static final String QUERY = "query";
+  public static final String STARTTIME = "starttime";
+  public static final String ENDTIME = "endtime";
+  public static final String STEP = "step";
+
+  /**
+   * Add table functions specific to Prometheus to repository.
+   */
+  public void register(BuiltinFunctionRepository repository) {
+    repository.register(queryRange());
+  }
+
+  private static FunctionResolver queryRange() {
+    FunctionName functionName = BuiltinFunctionName.QUERY_RANGE.getName();
+    FunctionSignature functionSignature =
+        new FunctionSignature(functionName, List.of(STRING, LONG, LONG, LONG));
+    final List<String> argumentNames = List.of(QUERY, STARTTIME, ENDTIME, STEP);
+
+    FunctionBuilder functionBuilder = arguments -> {
+      Boolean argumentsPassedByName = arguments.stream()
+          .noneMatch(arg -> StringUtils.isEmpty(((NamedArgumentExpression) arg).getArgName()));
+      Boolean argumentsPassedByPosition = arguments.stream()
+          .allMatch(arg -> StringUtils.isEmpty(((NamedArgumentExpression) arg).getArgName()));
+      if (!(argumentsPassedByName || argumentsPassedByPosition)) {
+        throw new SemanticCheckException("Arguments should be either passed by name or position");
+      }
```

Comment:
You need an additional check for a missing argument. This error message would be unclear in such case.